src/client/dfs/dfs_internal.h

```diff
@@ -53,6 +53,22 @@ dfs_lookupx(dfs_t *dfs, dfs_obj_t *parent, const char *name, int flags,
 	    dfs_obj_t **obj, mode_t *mode, struct stat *stbuf, int xnr,
 	    char *xnames[], void *xvals[], daos_size_t *xsizes);
 
+/* oid-in variants of dfs functions, these can be be used to verify the file being clobbered */
+int
+dfs_removeoi(dfs_t *dfs, dfs_obj_t *parent, const char *name, bool force, daos_obj_id_t *oid);
```

Comment:
removeoi is a weird name to me.
i would suggest:
dfs_remove_oid_verify or something like that. same for move

Reply:
I'll just make it dfs_remove_internal(), the benefit of internal functions is we are able to change their prototypes as we need so we don't need to embed the specifics of how it varies in the name,